contracts/ProjectValidation.sol

```diff
@@ -0,0 +1,180 @@
+pragma solidity ^0.4.10;
+
+import "zeppelin-solidity/contracts/token/SimpleToken.sol";
+import "zeppelin-solidity/contracts/math/SafeMath.sol";
+
+
+contract ProjectValidation {
+
+    enum Stages {
+        collectingSignatures,
+        checkerWork,
+        readyToComplete,
+        closing,
+        successfullyClosed,
+        unsuccessfullyClosed
+    }
+
+    struct Checker {
+        address addr;
+        bool signature;
+        bool signed;
+        bool presence;
+    }
+
+    address public starter;
+
+    address public manager;
+
+    Checker public checker;
+
+    address public exchangerContract;
+
+    bool public projectClosed;
+
+    bool public projectCompleted;
+
+    uint public additionalTokenRate;
+
+    uint public projectBalance;
+
+    uint public workerRatio = 2;
+
+    uint public amountForRoots = 0;
+
+    SimpleToken public fundTokens;
+
+    address[] public workers;
+
+    mapping (address => uint) public workersBalances;
+
+    mapping (address => bool) public signatures;
+
+    event Singed(address who);
+    event Closed();
+    event StateChanged(Stages previous, Stages current);
+
+    Stages public stage = Stages.collectingSignatures;
+
+    function ProjectValidation(
+        address _manager,
+        address _checker,
+        address _exchangerContract,
+        address[] _workers,
+        address fundTokenAddress
+    ) {
+        starter = msg.sender;
+        manager = _manager;
+        checker = Checker(
+            _checker,
+            false,
+            false,
+            false
+        );
+        exchangerContract = _exchangerContract;
+        fundTokens = SimpleToken(fundTokenAddress);
+        workers = _workers;
+        signatures[starter] = false;
+        signatures[manager] = false;
+        for (uint8 i = 0; i < workers.length; i++) {
+            workersBalances[workers[i]] = 0;
+        }
+    }
+
+    modifier onlyValidator() {
+        require(msg.sender == manager || msg.sender == starter || msg.sender == checker.addr);
+        _;
+    }
+
+    modifier onlyChecker(){
+        require(msg.sender == checker.addr && checker.presence);
+        _;
+    }
+
+    modifier atStage(Stages _stage) {
+        require(stage == _stage);
+        _;
+    }
+
+    function changeStateTo (Stages _stage) internal {
+        StateChanged(stage, _stage);
+        stage = _stage;
+    }
+
+    modifier afterExecutingGoToState(Stages _stage){
+        _;
+        changeStateTo(_stage);
+    }
+
+    function sign() onlyValidator atStage(Stages.collectingSignatures) returns (bool signed) {
+        signatures[msg.sender] = true;
+        return true;
+    }
+
+    function checkerSign(bool signature) onlyChecker atStage(Stages.checkerWork) afterExecutingGoToState(Stages.readyToComplete) {
+        checker.signed = true;
+        checker.signature = signature;
+    }
+
+    function checkerCall() internal {
```

Comment:
this functions used only once.
mb it is better to inline those 2 lines?

Reply:
yes, you are right, I will write this inline;